ocs_ci/ocs/ui/base_ui.py

```diff
@@ -1485,20 +1500,1136 @@ def __init__(self):
+
+        Example:
+            get_nested_deployment_names_of_node_from_df_ui("my-node")
+            # Returns ['deployment-1', 'deployment-2', 'deployment-3']
+        """
+        return list(
+            self.get_nested_deployments_of_node_from_df_ui(node_name)["entity_name"]
+        )
+
+    def update_topology_status(self):
+        """
+        Updates the status of entities in the topology.
+
+        This method iterates over the rows of the topology dataframe and updates the entity status, status XPath,
+        select node XPath, and navigate into XPath (if applicable) for each entity.
+
+        Note:
+            The method assumes the availability of the topology dataframe, certain locators,
+            and a brief pause of 0.1 seconds between iterations.
+
+        """
+
+        from ocs_ci.ocs.ui.helpers_ui import format_locator
+
+        df = self.topology_df
+        for index, row in df.iterrows():
+            entity_name = row["entity_name"]
+            df.loc[
+                df["entity_name"] == entity_name, "entity_status"
+            ] = self._get_status_of_entity(entity_name)
+            df.loc[df["entity_name"] == entity_name, "status_xpath"] = format_locator(
+                self.topology_loc["node_status_class_axis"], entity_name
+            )[0]
+            df.loc[
+                df["entity_name"] == entity_name, "select_node_xpath"
+            ] = format_locator(self.topology_loc["select_entity"], entity_name)[0]
+            # navigate_into_xpath is applicable only for node level, since we can not navigate into deployment
+            if "navigate_into_xpath" in df.columns:
+                df.loc[
+                    df["entity_name"] == entity_name, "navigate_into_xpath"
+                ] = format_locator(
+                    self.topology_loc["enter_into_entity_arrow"], entity_name
+                )[
+                    0
+                ]
+            time.sleep(0.1)
+
+    def initiate_topology_df(self, reinit: bool = True):
+        """
+        Initializes the topology DataFrame if not already set or if explicitly requested.
+
+        Args:
+            reinit (bool, optional): Whether to reinitialize the topology DataFrame. Defaults to True.
+
+        Raises:
+            NoSuchElementException: If the element text cannot be read.
+
+        Note:
+            The method assumes the availability of the self.topology_df DataFrame and certain constants.
+
+        """
+        if not self.topology_df["entity_name"].notna().any() or reinit:
+            entities = self.get_elements(self.topology_loc["node_label"])
+            entity_names = []
+            for entity in entities:
+                text = entity.text
+                if not len(text):
+                    raise NoSuchElementException("Cannot read element text")
+                name = text.split("\n")[1]
+                entity_names.append(name)
+                time.sleep(0.1)
```

Comment:
Why do you need to use in sleep in this function?
IIUC you don't interact with UI...